### ISSUE TITLE
Enum base class member type bug

### ISSUE THREAD

**Issue by ivanthewebber** [python/mypy#20461]
I am wanting to use a base class to stand in for StrEnum as we are currently on 3.10 but plan to upgrade to 3.14 soon. However, when running MyPy on Python 10 it wrongly thinks enum members are str.

**Bug Report**

(A clear and concise description of what the bug is.)

**To Reproduce**
[mypy-play](https://mypy-play.net/?mypy=latest&python=3.10&gist=df7ed5728a941d3059a21a42c02374c7)

**Expected Behavior**
No error on assignment to `a2`.

**Actual Behavior**
False error.

**Your Environment**

- Mypy version used: latest
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.10 (pre 11)

**Comment by ivanthewebber** [comment]
Fixed the link

**Comment by ivanthewebber** [comment]
As a workaround, moving the class out of the conditional seems to work:
```python
from enum import Enum
from sys import version_info


class IntEnum(int, Enum):
    """Placeholder `IntEnum` for Python 3.11+ compatibility.

    Python 3.11 changed enum string and mix-in behavior for `MyEnum(int, Enum)`:
    - Python 3.10: f"{Codes.bad}" → "400"
    - Python 3.11: f"{Codes.bad}" → "Codes.bad"

    IntEnum in 3.11+ restores the old behavior.
    """


class StrEnum(str, Enum):
    """Placeholder `StrEnum` for Python 3.11+ compatibility.

    Python 3.11 changed enum string and mix-in behavior for `MyEnum(str, Enum)`:
    - Python 3.10: f"{RouteNames.connect}" → "connect"
    - Python 3.11: f"{RouteNames.connect}" → "RouteNames.connect"

    StrEnum in 3.11+ restores the old behavior.
    """


if version_info.minor >= 11:
    # TODO once stable on 3.14 just use these directly
    from enum import IntEnum, StrEnum  # type: ignore
```

**Comment by sterliakov** [comment]
Mypy only understands some version guards, and partial attribute access is not one of them. For `sys.version_info` check to be understood by mypy, it should look like this (fully-qualified `sys.version_info` plus tuple comparison):

```python
if sys.version_info >= (3, 11): ...
```

If you switch to such supported version guard, your code passes on both 3.10 and 3.13: https://mypy-play.net/?mypy=latest&python=3.10&gist=c665c80ed63a3ea13ca9d0bab67e55ba

In general, mypy doesn't really support conditional class definitions, let alone doing that together with complex enum semantics - you can only reasonably use them with supported [version/platform checks](https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks). The important errors in your case are

```
main.py:6: error: Module "enum" has no attribute "StrEnum"  [attr-defined]
main.py:8: error: Name "StrEnum" already defined (possibly by an import)  [no-redef]
```

and the diagnostic on assignment is just a consequence of that. The same happens if you write just

```python
if foo:
    class A: ...
else:
    class A: ...
```
